src/Services/Search/ProviderAwareRequestProxyFactory.php

```diff
@@ -0,0 +1,131 @@
+    {
+        // Remove prefixed /search from our routes
+        $originalUri = $request->getUri();
+        $searchPath = \mb_substr($originalUri->getPath(), 7);
+
+        // Build provider specific search path.
+        $searchPath = $this->buildProviderSpecificPath($provider, $searchPath);
+
+        // Rewrite the Search Uri with our Elasticsearch host and new search path
+        // Note: the $searchUri is also how we define authorisation to be used for this proxying.
+        $searchUri = (new Uri($this->elasticHost))
+            ->withPath($searchPath)
+            ->withQuery($originalUri->getQuery())
+            ->withFragment($originalUri->getFragment());
+
+        // Builds a new request target
+        $requestTarget = $this->buildRequestTarget($request, $searchPath);
+
+        if (($request instanceof ServerRequestInterface) === true) {
+            // If we get a ServerRequest, remove _encoder added by our framework
+
+            /**
+             * @var \GuzzleHttp\Psr7\ServerRequest $request
+             *
+             * @see https://youtrack.jetbrains.com/issue/WI-37859 - typehint required until PhpStorm recognises === chec
+             */
+            $request = $request->withoutAttribute('_encoder');
+        }
+
+        // Strip any authentication headers from the request.
+        $request = $request->withoutHeader('Authorization');
+
+        $userInfo = $searchUri->getUserInfo();
+        if ($userInfo !== '') {
+            // If the URI had a userinfo component, add an Authorization header.
+
+            $request = $request->withHeader(
+                'Authorization',
+                \sprintf('Basic %s', \base64_encode($userInfo))
+            );
+        }
+
+        return $request
+            // Remove the original Content-Length - the underlying Http Client will repopulate
+            ->withoutHeader('Content-Length')
+            // Force a json content type
+            ->withHeader(
+                'Content-Type',
+                'application/json'
+            )
+            // Swap out the original request target with our new one
+            ->withRequestTarget($requestTarget ?: '/')
+            // Replace the URL
+            ->withUri($searchUri);
+    }
+
+    /**
+     * Build provider specific search path.
+     *
+     * @param \LoyaltyCorp\Multitenancy\Database\Entities\Provider $provider Provider entity
+     * @param string $searchPath Original search path
+     *
+     * @return string Provider specific search path
+     */
+    private function buildProviderSpecificPath(Provider $provider, string $searchPath): string
+    {
+        $pathtokens = \explode('/', $searchPath);
+
+        // Here we are assuming that the first part of the path will be index name.
+        if (\count($pathtokens) >= 2) {
+            $pathtokens[1] = \sprintf('%s_%s', $pathtokens[1], \mb_strtolower($provider->getExternalId()));
```

Comment:
Sorry to lump this on you, but elasticsearch supports searching multiple indicies at once and it isnt something we've considered before but will want to handle in this class.

It is possible to make a query like this:

`/search/customers,transactions/_search` which will run the search against both the customers and transactions indicies and they'll both need to be transformed into `customers_providerid,transactions_providerid`

Reply:
Ugh. Can of worms. 

Lets change this system to ensure that there is only a single non comma separated string that is not _search or _all.

Reply:
added a check to make sure the first part is not _all or _search else throw. and then splits index on `,` and adds provider id to all.